vortex-scalar/src/lib.rs

```diff
@@ -140,6 +140,33 @@ impl Scalar {
             value: self.value,
         }
     }
+
+    /// Size of the scalar in bytes, uncompressed.
+    pub fn nbytes(&self) -> usize {
+        match self.dtype() {
+            DType::Null => 0,
+            DType::Bool(_) => 1,
+            DType::Primitive(ptype, _) => ptype.byte_width(),
+            DType::Binary(_) | DType::Utf8(_) => self
+                .value()
+                .as_buffer()
+                .ok()
```

Comment:
Why does as_buffer return a result? as_struct doesn't. Maybe as_buffer_opt should

Reply:
honestly can just return an `Option`, the failure cases are pretty simple. Mind I do it in a followup?